src/web/src/main.js

```diff
@@ -13,10 +14,24 @@ Vue.config.productionTip = false;
 
 Vue.use(VueRouter);
 Vue.use(VueCookies);
+Vue.use(Vuex);
+
+const store = new Vuex.Store({
+  state: {
+    darkMode: false,
+  },
+  mutations: {
+    setDarkMode(state, bool) {
```

Comment:
I'd also suggest using constants for mutation and action types. Here's the official docs on it: https://vuex.vuejs.org/guide/mutations.html#using-constants-for-mutation-types. You don't need to put the constants in a separate file, having them at the top of the `index.js` will work fine